plugins/src/lib.rs

```diff
@@ -57,11 +57,11 @@ pub struct ConfiguredPlugin<S, I, O>
 where
     S: Clone + Send,
 {
+    pub configuration: Configuration,
```

Comment:
nit: `Plugin` exposes this through a method instead of making it public, should `ConfiguredPlugin` do so in the same way?

Reply:
not sure what is the benefit of this, but yeah I will add

Reply:
Well, just design consistency. You normally either have a public field or a getter.

Up to you, it just felt weird to see both.

Reply:
No no, you are right! we must follow the same API otherwise we make the stuff messy! also in this way, the configuration can not be mutated.